docs/how-to/define-state-machines.md

```diff
@@ -0,0 +1,427 @@
+
+---
+
+## Prerequisites
+
+Before starting, you should have:
+
+* Completed the [Hello World Tutorial](https://fprime.jpl.nasa.gov/latest/tutorials-hello-world/docs/hello-world/) (so you’ve built and run at least one component).
+* A general understanding of [FPP component modeling](https://nasa.github.io/fpp/fpp-users-guide.html).
+* Experience creating commands, events, and telemetry in FPP.
+* A working build of F Prime on your system (`fprime-util` runs successfully).
+
+---
+
+## When to Use State Machines
+
+State machines are useful when your component has distinct **modes** or **operational states** with different behavior in specific states and rules for the transitions between states. Examples include:
+
+* A radio with `OFF`, `IDLE`, and `TRANSMITTING` states.
+* A sensor with a series of start-up states
+
+By modeling these as a state machine, you make system behavior explicit, verifiable, and easier to test.
+
+---
+
+## Example State Machine
+
+This guide will walk through the steps of implementing a state machine with the following properties:
+
+0. The initial state is `RESET`. Followed by `WAIT_RESET`, `ENABLE`, `CONFIGURE` and `RUN`.
+1. At each state a `tick` signal will trigger an action.
+2. Actions may yield `success` or `error` signals.
+3. `success` signals progress linearly through to the next state.
+4. `error` signals return to state `RESET`
+5. The state machine remains in a given state (and repeats the action) when the action yields no signals.
+
+This state machine can be seen in the following diagram.
+
+```mermaid
+stateDiagram-v2
+    [*] --> RESET
+
+    %% State definitions with annotations
+    state "RESET
+
+    tick / doReset" as RESET
+
+    state "WAIT_RESET
+
+    tick / checkReset" as WAIT_RESET
+
+    state "ENABLE
+
+    tick / doEnable" as ENABLE
+
+    state "CONFIGURE
+
+    tick / doConfigure" as CONFIGURE
+
+    state "RUN
+
+    tick / doRead" as RUN
+
+    RESET --> WAIT_RESET : success
+    WAIT_RESET --> ENABLE : success
+    ENABLE --> CONFIGURE : success
+    CONFIGURE --> RUN : success
+
+    ENABLE --> RESET : error
+    WAIT_RESET --> RESET : error
+    CONFIGURE --> RESET : error
+    RUN --> RESET : error
+```
+
+This state machine is implemented in the [MpuImu](TODO) component. This component drives the "next" signal from a rate group handler.
+
+> [!NOTE]
+> The [MpuImu](TODO) component implements a further `reconfigure` signal allowing the machine to return to reconfigure state.
+
+## Designing a State Machine in FPP
+
+We will model the Imu state machine in FPP.  Typically users would define state machines in a new file.  We should use `ImuStateMachine.fpp` defined in the same folder (i.e. module) as the `ImuManager` this will support.  Define the basic module and state machine with:
```

Comment:
>Typically users would define state machines in a new file.

This statement seems a bit strong. In my opinion:

* If the state machine is part of the `ImuManager` component implementation, the standard practice is to define it inside the component.
* If it's a more general state machine for use by multiple components, then it would normally go in its own directory.

Using a separate `.fpp` file in the same directory as the component is possible but doesn't seem standard. Maybe at least mention the other options.